### DISCUSSION TITLE
SKR2 FW update via SD card

### DISCUSSION THREAD

**Discussion by boboxx** [grblHAL/STM32F4xx#156] [General]
I have a SKR2 board that everytime I want to update the FW, I have to install via the ST-Link. To resolved that I reinstalled the boot loader from BigTree and tried to load grblhal via SD card but for some reason it wont load. If I swap out the FW file on the SD card for Marlin it loads no problem. I see the file being renamed as firmware.cur so I know the bootloader is processing the file. I have tried it on a F429 and a F407. Both bootloader FW came from BigTree support.

I'm using the web builder to generate the FW, do I have to do anyting special to be able to load via the SD card?

**Comment by boboxx** [comment]
Using the bootloader address defined here for the SKR2 I have the same behaviour if I load the boot loader and then I tried to flash the grblhal FW from 0x800800

https://github.com/GadgetAngel/BTT_SKR_13_14_14T_SD-DFU-Bootloader/blob/main/bootloader_bin/backed_up_original_bootloaders/SKR%20V2.0%20(SKR%202)/Bigtreetech%20boards%20Bootloaders_V6.5.pdf


if i load the grblhal FW from 0x800000 all is good except I can't use the boot loader to update

**Reply by terjeio** [reply]
I have added a build option for a bootloader version, please try that and report back.
